artifactory/commands/transferfiles/api.go

```diff
@@ -108,12 +107,18 @@ type UuidTokenResponse struct {
 	UuidToken string `json:"uuid_token,omitempty"`
 }
 
-// Fill tokens batch till full. Return if no new tokens are available.
-func fillTokensBatch(awaitingStatusChunksSet *datastructures.Set[string], uploadTokensChan chan string) {
-	for awaitingStatusChunksSet.Size() < GetThreads() {
+// Fill chunk data batch till full. Return if no new chunk data is available.
+func fillChunkDataBatch(chunksResilientManager *ChunksResilientManager, uploadChunkChan chan UploadedChunkData) {
+	for chunksResilientManager.numChunksInMap < GetThreads() {
 		select {
-		case token := <-uploadTokensChan:
-			awaitingStatusChunksSet.Add(token)
+		case data := <-uploadChunkChan:
+			if _, exist := chunksResilientManager.nodeToChunksMap[data.NodeId]; !exist {
+				chunksResilientManager.nodeToChunksMap[data.NodeId] = map[string][]FileRepresentation{}
+			}
+			if _, exist := chunksResilientManager.nodeToChunksMap[data.NodeId][data.ChunkUuid]; !exist {
```

Comment:
I think that we can even delete the second if statement, since in this function we are only getting new UUIDs, therefore it's impossible to have this UUID in the map already, so the if statement will always be true.